File: app/src/main/java/nu/parley/fcm/FirebaseMessagingService.java
package nu.parley.fcm;

import android.content.Intent;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.RemoteMessage;

import nu.parley.android.Parley;
import nu.parley.ui.ChatActivity;

public final class FirebaseMessagingService extends com.google.firebase.messaging.FirebaseMessagingService {

    @Override
    public void onNewToken(@NonNull String token) {
        super.onNewToken(token);

        Parley.setPushToken(token);
    }

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        Intent intent = new Intent(this, ChatActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);

        @SuppressWarnings("unused")
        boolean handledByParley = Parley.handle(this, remoteMessage.getData(), intent);
    }
}

File: app/src/main/java/nu/parley/fcm/FirebaseMessagingService.kt
package nu.parley.fcm

import android.content.Intent
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import nu.parley.android.Parley
import nu.parley.ui.ChatActivity

class FirebaseMessagingService : FirebaseMessagingService() {
    override fun onNewToken(token: String) {
        super.onNewToken(token)

        Parley.setPushToken(token)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)

        val intent = Intent(this, ChatActivity::class.java)
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)

        @Suppress("UNUSED_VARIABLE")
        val handledByParley = Parley.handle(this, remoteMessage.data, intent)
    }
}